jquery.jsticky.js

```diff
@@ -43,56 +43,75 @@
 
     return this.each(function() {
 
-      var $this          = $(this),
-          thisHeight     = $this.outerHeight(),
-          thisWidth      = $this.outerWidth(),
-          topSpacing     = settings.topSpacing,
-          zIndex         = settings.zIndex,
-          pushPoint      = $this.offset().top - topSpacing, // Point at which the sticky element starts pushing
-          placeholder    = $('<div></div>').width(thisWidth).height(thisHeight).addClass('sticky-placeholder'), // Cache a clone sticky element
-          stopper        = settings.stopper,
-          $window        = $(window);
+      var $this = $(this);
+      var thisHeight = $this.outerHeight();
+      var thisWidth = $this.outerWidth();
+      var topSpacing = settings.topSpacing;
+      var zIndex = settings.zIndex;
+      var pushPoint = $this.offset().top - topSpacing; // Point at which the sticky element starts pushing
+      var placeholder = $('<div></div>').width(thisWidth).height(thisHeight).addClass('sticky-placeholder'); // Cache a clone sticky element
+      var stopper = settings.stopper;
+      var $window = $(window);
 
       function stickyScroll() {
 
         var windowTop  = $window.scrollTop(); // Check window's scroll position
-        
+        var stopPoint = stopper;
+        var parentWidth = $this.parent().width();
+
+        placeholder.width(parentWidth)
+
         if ( hasStopper && typeof stopper === 'string' ) {
-          var stopperTop = $(stopper).offset().top,
-              stopPoint  = (stopperTop - thisHeight) - topSpacing;
-        } else if (hasStopper && typeof stopper === 'number') {
-          var stopPoint = stopper;
+          var stopperTop = $(stopper).offset().top;
+          stopPoint  = (stopperTop - thisHeight) - topSpacing;
         }
 
         if (pushPoint < windowTop) {
           // Create a placeholder for sticky element to occupy vertical real estate
+          if(settings.stickyClass)
+            $this.addClass(settings.stickyClass);
+
           $this.after(placeholder).css({
             position: 'fixed',
-            top: topSpacing
+            top: topSpacing,
+            width: parentWidth
```

Comment:
I would have thought you'd want it to occupy the same space as when it's not stuck? This is essentially what this does 😄